domains/runtime/evm/src/lib.rs

```diff
@@ -908,49 +958,98 @@ impl_runtime_apis! {
             }
         }
 
-        fn check_transaction_validity(
+        fn check_transaction_validity(uxt: &<Block as BlockT>::Extrinsic,
+            block_number: BlockNumber,
+            block_hash: <Block as BlockT>::Hash) -> Result<(), domain_runtime_primitives::CheckTxValidityError>  {
+            let lookup = frame_system::ChainContext::<Runtime>::default();
+            check_transaction_validity_inner(&lookup, uxt, block_number, block_hash).map(|_| ())
+        }
+
+        fn check_transaction_validity_and_do_pre_dispatch(
```

Comment:
While we do need to perform `do_pre_dispatch` here for `CheckNonce` to increase the account nonce (perhaps also for `CheckWeight`), both the `fn validate` and `fn pre_dispatch` of the `ChargeTransactionPayment` will deduct the tx fee from the tx signer which means it is checking if the user can pay `2*tx_fee` instead of `tx_fee`.

One option would be to use a customized `CheckNonce` which will increase nonce in `fn validate` and not perform the `pre_dispatch` here, but something more generic will be preferred, WDYT @vedhavyas

Reply:
@NingLin-P Hmm. This is concerning. No need to modify `CheckNonce`. Let me take a look, I do have an idea.

Reply:
Make sense to me to only perform `pre_dispatch`. Per document, `pre_dispatch` is suggested to perform the same checks as in `validate`.